pgjdbc/src/main/java/org/postgresql/core/Encoding.java

```diff
@@ -12,70 +12,95 @@
 import java.io.OutputStreamWriter;
 import java.io.Reader;
 import java.io.Writer;
+import java.nio.charset.Charset;
 import java.util.HashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /**
  * Representation of a particular character encoding.
  */
 public class Encoding {
-  private static final Encoding DEFAULT_ENCODING = new Encoding(null);
+
+  private static final Logger LOGGER = Logger.getLogger(Encoding.class.getName());
+
+  private static final Encoding DEFAULT_ENCODING = new Encoding();
+  private static final Encoding UTF8_ENCODING = new Encoding("UTF-8");
 
   /*
    * Preferred JVM encodings for backend encodings.
    */
-  private static final HashMap<String, String[]> encodings = new HashMap<String, String[]>();
+  private static final HashMap<String, String[]> ENCODINGS = new HashMap<String, String[]>();
```

Comment:
paraphrasing google style guide  "merely intending not to change it is not enough"